arcgis-ios-sdk-samples/Layers/Display WFS/DisplayWFSViewController.swift

```diff
@@ -0,0 +1,98 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class DisplayWFSViewController: UIViewController {
+
+    @IBOutlet private weak var mapView: AGSMapView!
+    private var wfsFeatureTable : AGSWFSFeatureTable!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Initialize a map with topographic basemap
+        let map = AGSMap(basemap: .topographic())
+        
+        // A URL to the GetCapabilities endpoint of a WFS service
+        let wfsServiceURL = URL(string: "https://dservices2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/services/Seattle_Downtown_Features/WFSServer?service=wfs&request=getcapabilities")!
+        
+        // Name of the table to load
+        let wfsTableName = "Seattle_Downtown_Features:Buildings"
+        
+        // Initialize a WFS feature table with service URL and uniquely identifying table name
+        wfsFeatureTable = AGSWFSFeatureTable(url: wfsServiceURL, tableName: wfsTableName)
+        
+        // Set feature request mode to manual - only manual is supported at v100.5.
+        // In this mode, you must manually populate the table - panning and zooming won't request features automatically.
+        wfsFeatureTable.featureRequestMode = .manualCache
+    
+        // Initialize a feature layer from WFS feature table
+        let wfsFeatureLayer = AGSFeatureLayer(featureTable: self.wfsFeatureTable)
+        
+        // Create a simple renderer
+        let simpleLineSymbol = AGSSimpleLineSymbol(style: .solid, color: .red, width: 3.0)
+        let renderer = AGSSimpleRenderer(symbol: simpleLineSymbol)
+        
+        // Apply renderer to feature layer
+        wfsFeatureLayer.renderer = renderer
+        
+        // Add feature layer to the map
+        map.operationalLayers.add(wfsFeatureLayer)
+        
+        // Query for features whenever map's viewpoint changes
+        self.mapView.viewpointChangedHandler = { [weak self] in
+            DispatchQueue.main.async {
+                self?.populateFeaturesFromQuery()
+            }
+        }
+        
+        // Set initial viewpoint
+        map.initialViewpoint = AGSViewpoint(targetExtent: AGSEnvelope(xMin: -122.341581, yMin: 47.613758, xMax: -122.332662, yMax: 47.617207, spatialReference: .wgs84()))
+        
+        // Assign the map to the map view
+        mapView.map = map
+
+        // Add the source code button item to the right of navigation bar
+        (navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["DisplayWFSViewController"]
+    }
+    
+    private func populateFeaturesFromQuery() {
+        
+        // Create query parameters
+        let params = AGSQueryParameters()
+        params.geometry = self.mapView.visibleArea?.extent
+        params.spatialRelationship = .intersects
+        
+        // Show progress
+        SVProgressHUD.show(withStatus: "Querying")
```

Comment:
This is raising a flag with Main thread checker with following warning

`Main Thread Checker: UI API called on a background thread: -[UIApplication applicationState]`

Reply:
Hmm, I'm not sure why that would be. There are other samples that use `SVProgressHUD`, is the warning present in only this one?

Reply:
Actually I am seeing this in some other samples where we trigger progress HUD this from a completion handler. Maybe we can handle all these in a different issue? what do you think @philium ?

Reply:
I am not able to reproduce the the warning with this sample. But if it is happening in other samples, I would definitely create an issue for it. As to whether we can create a one-size-fits-all solution, I don't know. Maybe? I'd have to think about it.